Dynatrace/dynatrace-operator
# Implements `automatic-kubernetes-api-monitoring-cluster-label` feature flag

# Description
`AutomaticKubernetesApiMonitoring` feature uses `clusterLabel` to show it on UI.
By default `clusterLabel` is set to `dynakube` CR name.
If `feature.dynatrace.com/automatic-kubernetes-api-monitoring-cluster-label` annotation is set, it's value is used to override default value of `clusterLabel`.

Unicode characters are supported as a value.

## How can this be tested?
```
kubectl create namespace dynatrace
kubectl apply -f https://github.com/Dynatrace/dynatrace-operator/releases/download/v0.6.0/kubernetes.yaml
kubectl -n dynatrace wait pod --for=condition=ready --selector=app.kubernetes.io/name=dynatrace-operator,app.kubernetes.io/component=webhook --timeout=300s
kubectl apply -f - <<EOF
apiVersion: v1
data:
  apiToken: XXXX
  dataIngestToken: XXXX
kind: Secret
metadata:
  name: dk-4
  namespace: dynatrace
type: Opaque
---
apiVersion: dynatrace.com/v1beta1
kind: DynaKube
metadata:
  name: dk-4
  namespace: dynatrace
  annotations:
    feature.dynatrace.com/automatic-kubernetes-api-monitoring: "true"
    feature.dynatrace.com/automatic-kubernetes-api-monitoring-name: "...dots;;;źdźbło~!🙃."
spec:
  apiUrl: https://xxxxx/api
  activeGate:
    capabilities:
      - routing
      - kubernetes-monitoring
EOF
```
check if UI shows:
![image](https://user-images.githubusercontent.com/35979436/170028823-b87dc2d1-6d61-4ea7-97d0-50661c2b15de.png)


## Checklist
